dll/win32/shell32/shlfolder.cpp

```diff
@@ -274,6 +274,42 @@ HRESULT SHELL32_CompareDetails(IShellFolder2* isf, LPARAM lParam, LPCITEMIDLIST
     return MAKE_COMPARE_HRESULT(ret);
 }
 
+HRESULT SHELL_CompareAllFields(IShellFolder* psf,
+                                 LPCITEMIDLIST pidl1, LPCITEMIDLIST pidl2,
+                                 UINT Count, int Skip)
```

Comment:
```suggestion
                               LPCITEMIDLIST pidl1, LPCITEMIDLIST pidl2,
                               UINT Count, int Skip)
```

Reply:
Why "Skip" needs to be "int" ?

Reply:
Needs is a strong word but it hints the caller can pass -1 if there is no column to skip. It is up to the folder implementation to know how SHCIDS_CANONICALONLY influences how the canonical column is compared (not yet handled in ROS).